pkg/component/certmanagement/deployment.go

```diff
@@ -42,8 +42,8 @@ const (
 	rsaPrivateKeySize = 3072
 )
 
-// newCertManagementDeployment creates a new instance of DeployWaiter for the CertManagement deployment.
-func newCertManagementDeployment(cl client.Client, values Values) component.DeployWaiter {
+// NewDeployment creates a new instance of DeployWaiter for the CertManagement controller.
+func NewDeployment(cl client.Client, values Values) component.DeployWaiter {
```

Comment:
Nit
```suggestion
func NewController(cl client.Client, values Values) component.DeployWaiter {
```
According to the description and the name of the component in gardener-operator reconciler. You could also rename the file.